requests/models.py

```diff
@@ -420,8 +420,11 @@ def prepare_body(self, data, files, json=None):
         length = None
 
         if not data and json is not None:
-            content_type = 'application/json'
-            body = complexjson.dumps(json)
+            # When urllib3 uses pyOpenSSL, it can only resume large uploads
+            # properly if receiving a bytes-like object. In Python 2, json.dumps()
+            # returns just that, but Python 3 returns a Unicode string.
+            content_type = 'application/json; charset=utf-8'
+            body = complexjson.dumps(json).encode('utf-8')
```

Comment:
This is too aggressive: it encodes to UTF-8 even when `json` returns a bytestring, which it does on Python 2: that means that this line of code can cause UnicodeDecodeErrors when run if the user's default encoding is no good. I think we need a test on the type here.